spec/datadog/tracing/contrib/utils/quantization/http_spec.rb

```diff
@@ -3,6 +3,18 @@
 require 'datadog/tracing/contrib/utils/quantization/http'
 
 RSpec.describe Datadog::Tracing::Contrib::Utils::Quantization::HTTP do
+  describe 'RFC3986_URL_BASE' do
+    let(:regex) { described_class::RFC3986_URL_BASE }
+
+    it 'matches in linear time' do
+      if !Regexp.respond_to?(:linear_time?)
+        skip 'Regexp.linear_time? method is only available on Ruby 3.3+'
+      else
+        expect(Regexp.linear_time?(regex)).to be_true
+      end
+    end
```

Comment:
Minor: I would suggest to actually match on the Ruby version, not on the method existing. 

That's because that would mean there are some Ruby versions that we basically assert that the method must exist, and would fail if it doesn't; whereas this version is somewhat weaker -- if I remove or hide the method on every Ruby version (maybe by mistake? maybe as a side-effect of another spec?), it's possible for our test suite to pass in CI and still the method not have been executed.

It's a small difference but, yeah, worth considering in general ;)

Reply:
Hmm, fair enough. I'm used to do that because of things like:

- https://github.com/marcandre/backports
- https://github.com/sqreen/ruby-backport

I'll amend though, not the least because it makes it clearer about the version.

Also I seem to have gotten it wrong and it's available on 3.2 as well?

Reply:
I normally do `obj.respond_to?(:blah) # Method available in Ruby 2.7+` 🤷

Reply:
> Also I seem to have gotten it wrong and it's available on 3.2 as well?

Something is off -- indeed 3.2 has `Regexp.linear_time?` and I'm getting this failure locally on 3.2

```Failures:

  1) Datadog::Tracing::Contrib::Utils::Quantization::HTTP RFC3986_URL_BASE matches in linear time
     Failure/Error: expect(Regexp.linear_time?(regex)).to be_true
       expected false to respond to `true?` or perhaps you meant `be true` or `be_truthy`
     # ./spec/datadog/tracing/contrib/utils/quantization/http_spec.rb:13:in `block (3 levels) in <top (required)>'
     # ./spec/spec_helper.rb:233:in `block (2 levels) in <top (required)>'
     # ./spec/spec_helper.rb:118:in `block (2 levels) in <top (required)>'
     # /home/ivo.anjo/.rvm/gems/ruby-3.2.5/gems/webmock-3.13.0/lib/webmock/rspec.rb:37:in `block (2 levels) in <top (required)>'
     # /home/ivo.anjo/.rvm/gems/ruby-3.2.5/gems/rspec-wait-0.0.10/lib/rspec/wait.rb:47:in `block (2 levels) in <top (required)>'

  2) Datadog::Tracing::Contrib::Utils::Quantization::HTTP OBFUSCATOR_REGEX matches in linear time
     Failure/Error: expect(Regexp.linear_time?(regex)).to be_true
       expected true to respond to `true?` or perhaps you meant `be true` or `be_truthy`
     # ./spec/datadog/tracing/contrib/utils/quantization/http_spec.rb:566:in `block (3 levels) in <top (required)>'
     # ./spec/spec_helper.rb:233:in `block (2 levels) in <top (required)>'
     # ./spec/spec_helper.rb:118:in `block (2 levels) in <top (required)>'
     # /home/ivo.anjo/.rvm/gems/ruby-3.2.5/gems/webmock-3.13.0/lib/webmock/rspec.rb:37:in `block (2 levels) in <top (required)>'
     # /home/ivo.anjo/.rvm/gems/ruby-3.2.5/gems/rspec-wait-0.0.10/lib/rspec/wait.rb:47:in `block (2 levels) in <top (required)>'

Finished in 0.09054 seconds (files took 0.40868 seconds to load)
228 examples, 2 failures

Failed examples:

rspec ./spec/datadog/tracing/contrib/utils/quantization/http_spec.rb:9 # Datadog::Tracing::Contrib::Utils::Quantization::HTTP RFC3986_URL_BASE matches in linear time
rspec ./spec/datadog/tracing/contrib/utils/quantization/http_spec.rb:562 # Datadog::Tracing::Contrib::Utils::Quantization::HTTP OBFUSCATOR_REGEX matches in linear time

Randomized with seed 48414
```

But not in CI...? Are we not running this spec in CI? :eyes: